init/main.go

```diff
@@ -697,6 +705,57 @@ func checkIfInitrd() error {
 	return nil
 }
 
+func blkInfoMatchesArg(info *blkInfo, arg string) bool {
+	if strings.HasPrefix(arg, "UUID=") && info.uuid == strings.TrimPrefix(arg, "UUID=") {
+		return true
+	}
+	if strings.HasPrefix(arg, "LABEL=") && info.label == strings.TrimPrefix(arg, "LABEL=") {
+		return true
+	}
+
+	return false
+}
+
+func resume(devpath string) error {
```

Comment:
does this patch handles suspend-on-disk only? is there anything special need to be to handle suspend-to-ram functionality?

Reply:
This is purely for suspend-to-disk. Suspend-to-ram doesn't go through a reboot, so it's handled purely by the running kernel.